components/devtools/actors/preference.rs

```diff
@@ -37,25 +38,28 @@ impl Actor for PreferenceActor {
         stream: &mut TcpStream,
         _id: StreamId,
     ) -> Result<ActorMessageStatus, ()> {
-        let mut key = msg.get("value").unwrap().as_str().unwrap();
-
-        // Mapping to translate a Firefox preference name onto the corresponding Servo preference name
-        let pref_name_mapping: HashMap<&str, &str> =
-            [("dom.serviceWorkers.enabled", "dom.serviceworker.enabled")]
-                .iter()
-                .copied()
-                .collect();
-        if pref_name_mapping.contains_key(key) {
-            key = pref_name_mapping.get(key).unwrap();
-        }
+        if let Some(mut key) = msg.get("value").unwrap().as_str() {
+            // Mapping to translate a Firefox preference name onto the corresponding Servo preference name
+            let pref_name_mapping: HashMap<&str, &str> =
+                [("dom.serviceWorkers.enabled", "dom.serviceworker.enabled")]
+                    .iter()
+                    .copied()
+                    .collect();
+            if pref_name_mapping.contains_key(key) {
+                key = pref_name_mapping.get(key).unwrap();
+            }
 
-        let pref_value = pref_map().get(key);
-        Ok(handle_preference_value(
-            pref_value,
-            self.name(),
-            msg_type,
-            stream,
-        ))
+            let pref_value = pref_map().get(key);
+            Ok(handle_preference_value(
+                pref_value,
+                self.name(),
+                msg_type,
+                stream,
+            ))
+        } else {
+            warn!("PreferenceActor: handle_message: value is not a string");
+            return Ok(ActorMessageStatus::Ignored);
+        }
```

Comment:
I think it makes a bit more sense to have this as an early return rather than wrapping the entire logic of the function in the error condition check.

Reply:
done, thanks!